script/tool/lib/src/publish_plugin_command.dart

```diff
@@ -85,15 +93,24 @@ class PublishPluginCommand extends PluginCommand {
   final Print _print;
   final Stdin _stdin;
   // The directory of the actual package that we are publishing.
-  Directory _packageDir;
   StreamSubscription<String> _stdinSubscription;
 
   @override
   Future<Null> run() async {
     checkSharding();
+    final String package = argResults[_packageOption];
+    if (package == null) {
```

Comment:
The help said --package is ignored if --all is provided, but this is bailing early regardless.

Reply:
Returning after finishing the review... why is this being added here? Nothing in the PR uses it; it's unimplemented and untested.

Reply:
Sorry to cause confusion. This PR doesn't handle --all yet. And this PR still tries to keep the original functionality of the command. 

Because I wanted to move `_checkPackageDir` after the `_verifyRemote` check, so I moved this check out of `_checkPackageDir` to keep the validation step consistent. 

The test is covered by `requires a package flag`